starport-polling-app/index.md

```diff
@@ -0,0 +1,277 @@
+
+Sign in with one of the passwords printed in the console and try creating a poll. You should see a new object created and displayed above the form. You have successfully created an object and stored it on the blockchain!
+
+This, however, does not look and work exactly like we need. We should be able to add option fields (and store them as an array) and they should be displayed as interactive buttons.
+
+Let's take a look at some of the files modified by the `starport type` command.
+
+### `x/voter/types/TypePoll.go`
+
+This file contains definition of the `Poll` type. We can see that a poll has two fields (creator and ID), which will be created automatically, and two fields (title and options) defined by us. Since we want `Options` to be a list of string, **replace `string` with `[]string`**
+
+### `x/voter/types/MsgCreatePoll.go`
+
+This file defines a message that creates a poll.
+
+To write anything to a blockchain or perform any other state transition a client (web app in our case) makes an HTTP POST request with a title and options to [http://localhost:1317/voter/poll](http://localhost:1317/voter/poll) endpoint handler for which is defined in `x/voter/client/rest/txPoll.go`. The handler creates an unsigned transaction which contains an array of messages. The client then signs the transaction and sends it to [http://localhost:1317/txs](http://localhost:1317/txs). The application then processes the transaction by sending each message to a corresponding handler, in our case `x/voter/handlerMessageCreatePoll.go`. A handler then calls a `CreatePoll` function defined in `x/voter/keeper/poll.go` which writes the poll data into the store.
+
+Going back to `MsgCreatePoll.go`, we need to make options to be stored as a list instead of a string. Replace `Options string` with `Options []string` in `MsgCreatePoll` struct and `options string` with `options []string` in the arguments of `NewMsgCreatePoll` function.
+
+### `x/voter/client/rest/txPoll.go`
+
+Replace `Options string` with `Options []string` in `createPollRequest` struct.
+
+### `x/voter/client/cli/txPoll.go`
+
+A user can also interact with our application through a command line interface.
+
+```
+votercli tx voter create-poll "Text editors" "Emacs" "Vim" --from user1
+```
+
+This command will generate a transaction with "create poll" message, sign it using a private key of `user1` (one of two users created by default) and broadcast it to the blockchain.
+
+The only modification we need to make is to change a line that reads arguments from the console: `argsOptions := args[:len(args)-1]`. This will assume that all arguments after the first one represent a list of options.
+
+Now that we have made all the necessary changes to our app, let's take a look at the client-side application.
+
+```jsx
+starport serve
+```
+
+### `frontend/src/view/Index.vue`
+
+Since we don't need the default form component replace `<type-list />` with a new component `<poll-form />`
```

Comment:
```suggestion
Since we don't need the default form component replace `<type-list />` inside of `frontent/src/views/Index.vue` with a new component `<poll-form />` that will be created in a new file at `frontend/src/components/PollForm.vue`.
```